config/config.go

```diff
@@ -161,6 +166,174 @@ type AuthConfig struct {
 	AppendMetadataHeaders bool          `yaml:"append_metadata_headers,omitempty" json:"append_metadata_headers,omitempty"`
 }
 
+// EnvConfigs
+type EnvConfigs struct {
+	// A list of files containing environment config
+	ConfigFiles []string `yaml:"config_files,omitempty" json:"config_files,omitempty"`
+
+	// A list of environment configs
+	InlineConfigs []EnvironmentConfig `yaml:"inline_configs,omitempty" json:"inline_configs,omitempty"`
+}
+
+// EnvironmentConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type EnvironmentConfig struct {
+	// Unique ID of the environment config
+	ID string `yaml:"id" json:"id"`
+
+	// A list of proxy configs
+	ProxyConfigs []ProxyConfig
+}
+
+// ProxyConfig has the proxy configuration
+type ProxyConfig struct {
+	// Top-level basepath for the proxy config
+	Basepath string `yaml:"basepath,omitempty" json:"basepath,omitempty"`
+
+	// A list of Operations, names of which must be unique within the proxy config.
+	Operations []Operation `yaml:"operations,omitempty" json:"operations,omitempty"`
+}
+
+// An API Operation associates a set of rules with a set of request matching
+// settings.
+type Operation struct {
+	// Name of the operation. Unique within a APIRuntimeControlConfig.
+	Name string `yaml:"name" json:"name"`
+
+	// Authentication defines the AuthenticationRequirement
+	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" json:"authentication,omitempty"`
+
+	// ConsumerAuthorization defines the ConsumerAuthorization
+	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" json:"consumer_authorization,omitempty"`
+
+	// HTTP matching rules for this operation.
+	HTTPMatches []HTTPMatch `yaml:"http_match,omitempty" json:"http_match,omitempty"`
+
+	// Target server for this operation. This will be sent to Envoy
+	// for routing to the corresponding upstream cluster upon a successful
+	// authorization of the operation.
+	Target Target `yaml:"target" json:"target"`
+}
+
+// AuthenticationRequirement defines the authentication requirement.
+// Precisely one of JWT, Any and All should be set.
+type AuthenticationRequirement struct {
+	// JWT defines the JWTAuthentication
+	JWT JWTAuthentication `yaml:"jwt,omitempty" json:"jwt,omitempty"`
+
+	// Any contains a list of AuthenticationRequirements.
+	// A successful authentication requires one of them being satisfied.
+	Any []AuthenticationRequirement `yaml:"any,omitempty" json:"any,omitempty"`
+
+	// All contains a list of AuthenticationRequirements.
+	// A successful authentication requires all of them being satisfied.
+	All []AuthenticationRequirement `yaml:"all,omitempty" json:"all,omitempty"`
+}
+
+// JWTAuthentication defines the JWT authentication
+type JWTAuthentication struct {
+	// Name of this JWT Provider, unique within the Proxy.
```

Comment:
This feels more like a "Requirement" than a "Provider". I bring this up because we use the term `Provider` below in `JWTClaim` which looks odd when referring to this object.

Reply:
Changed it to requirement.